src/System.Net.Primitives/src/System/Net/Cookie.cs

```diff
@@ -240,22 +239,23 @@ public string Name
             }
             set
             {
-                if (String.IsNullOrEmpty(value) || !InternalSetName(value))
+                if (!InternalSetName(value))
                 {
-                    throw new CookieException(SR.Format(SR.net_cookie_attribute, "Name", value == null ? "<null>" : value));
+                    throw new CookieException(SR.Format(SR.net_cookie_attribute, NameAttributeName, value == null ? "<null>" : value));
                 }
             }
         }
 
+        private bool IsValidName(string name)
+        {
+            return !string.IsNullOrEmpty(name) && name[0] != '$' && name.IndexOfAny(ReservedToName) == -1;
+        }
+
         internal bool InternalSetName(string value)
         {
-            if (String.IsNullOrEmpty(value) || value[0] == '$' || value.IndexOfAny(ReservedToName) != -1)
-            {
-                _name = string.Empty;
-                return false;
-            }
-            _name = value;
-            return true;
+            bool valid = IsValidName(value);
+            _name = valid ? value : string.Empty;
+            return valid;
```

Comment:
Simplifies the code and makes it more readable. Extracting the function out means that it can be used in VerifySetDefaults